remote_vector_index_builder/core/common/models/index_builder/faiss/ivf_pq_build_cagra_config.py

```diff
@@ -0,0 +1,140 @@
+    # store the given number of records. Set this flag to `true` if you prefer
+    # to use as little GPU memory for the database as possible.
+    conservative_memory_allocation: bool = False
+
+    @staticmethod
+    def _validate_params(params: Dict[str, Any]) -> None:
+        """
+        Pre-validates IVFPQBuildCagraConfig configuration parameters before object creation.
+
+        Args:
+            params: Dictionary of parameters to validate
+
+        Raises:
+            ValueError: If any parameter fails validation
+        """
+        if "n_lists" in params:
+            if params["n_lists"] <= 0:
+                raise ValueError(
+                    "IVFPQBuildCagraConfig param: n_lists must be positive"
+                )
+        if "kmeans_n_iters" in params:
+            if params["kmeans_n_iters"] <= 0:
+                raise ValueError(
+                    "IVFPQBuildCagraConfig param: kmeans_n_iters must be positive"
+                )
+        if "kmeans_trainset_fraction" in params:
+            if not 0 < params["kmeans_trainset_fraction"] <= 1:
+                raise ValueError(
+                    "IVFPQBuildCagraConfig param: kmeans_trainset_fraction must be between 0 and 1"
+                )
+        if "pq_bits" in params:
+            if params["pq_bits"] not in [4, 5, 6, 7, 8]:
+                raise ValueError(
+                    "IVFPQBuildCagraConfig param: pq_bits must be one of [4, 5, 6, 7, 8]"
+                )
+        if "pq_dim" in params:
+            pq_dim = params["pq_dim"]
+            if pq_dim < 0:
+                raise ValueError(
+                    "IVFPQBuildCagraConfig param: pq_dim must be non-negative"
+                )
+
+            # Check pq_dim constraints based on pq_bits
+            pq_bits = params.get("pq_bits", 8)  # Use 8 if not specified
+            if pq_bits != 8 and pq_dim % 8 != 0:
+                raise ValueError(
+                    "IVFPQBuildCagraConfig param: When pq_bits is not 8, pq_dim must be a multiple of 8"
+                )
+
+    def to_faiss_config(self) -> faiss.IVFPQBuildCagraConfig:
+        """
+        Creates and configures the equivalent FAISS IVFPQBuildCagraConfig from the
+        IVFPQBuildCagraConfig core datamodel.
+
+        Returns:
+             A configured FAISS IVFPQBuildCagraConfig object with parameters for:
+            - kmeans training set fraction
+            - kmeans iteration count
+            - Product Quantization bits and dimensions
+            - Number of inverted lists (kmeans clusters)
+            - Memory allocation strategy
+        """
+
+        config = faiss.IVFPQBuildCagraConfig()
+        config.kmeans_trainset_fraction = self.kmeans_trainset_fraction
+        config.kmeans_n_iters = self.kmeans_n_iters
+        config.pq_bits = self.pq_bits
+        config.pq_dim = self.pq_dim
+        config.n_lists = self.n_lists
+        config.conservative_memory_allocation = self.conservative_memory_allocation
```

Comment:
there were other parameters which we need to set for cagra index. why I am not seeing those parameters. Parameter include spaceType, store_dataset.